File: pages/refactorings/replace-global-state-with-parameter/01-hidden-setting/After.scala
// Total of an order: the lines summed, plus a fixed service fee.
object After:
  val serviceFee = 300 // cents; still the program's default

  def fee(serviceFee: Int, subtotal: Int): Int =
    subtotal + serviceFee

  def total(lines: List[Int]): Int = fee(serviceFee, lines.sum)

File: pages/refactorings/replace-global-state-with-parameter/01-hidden-setting/After.hs
-- Total of an order: the lines summed, plus a fixed service fee.
module After where

serviceFee :: Int
serviceFee = 300 -- cents; still the program's default

fee :: Int -> Int -> Int
fee serviceFee subtotal = subtotal + serviceFee

total :: [Int] -> Int
total lines = fee serviceFee (sum lines)
